payment/utils.py

```diff
@@ -68,6 +73,51 @@ def send_confirmation_email(user, order_number):
     email.send()
 
 
+def register_user_verified_cohort(user, order):
+    """
+    Once a user has been verified, add it to a cohort.
+
+    The cohort to which the user should be added is given by the VERIFIED_COHORTS setting.
+    settings.VERIFIED_COHORTS is a list of (r"<course id regex>", "<cohort name>") tuples. Regular expressions
+    are matched agains the course ID. If at least one regular expression matches, registered users from this
+    course are added to the cohort with the highest priority. Cohorts are sorted by increasing priority.
+
+    If a course is associated to a cohort that doesn't exist or an empty cohort name, then the user is not
+    added to any cohort. Thus, to disable this feature for a specific course, set:
+
+        VERIFIED_COHORTS = [
+            ...
+            (r"<course id>", None),
+            ...
+        ]
+
+    To enable this feature for a particular course, but not for all others, write:
+
+        VERIFIED_COHORTS = [
+            (r".*", None)
+            (r"<course id>", "cohort name"),
+        ]
+
+    To disable this feature globally, set:
+
+        VERIFIED_COHORTS = []
+    """
+    course = get_course(order)
+    verified_cohort_name = None
+    for course_id_regex, cohort_name in getattr(settings, "VERIFIED_COHORTS", []):
+        if re.match(course_id_regex, course.key):
+            verified_cohort_name = cohort_name
```

Comment:
@regisb if you don't break out of the "for" loop, you will not use the first one that matches but the last one? You should also add a test to secure this behavior.

Reply:
> if you don't break out of the "for" loop, you will not use the first one that matches but the last one?

Yes, exactly. See the above comment: "If at least one regular expression matches, registered users from this course are added to the cohort with the highest priority. Cohorts are sorted by increasing priority."
With this procedure, the default cohort should be added to the first element of the VERIFIED_COHORTS setting. We could do the opposite, and then the default cohort would be last. This behaviour is tested in `test_register_user_custom_course_cohort`.

Reply:
oh ok I didn't realize the "increasing priority". I feel it's more natural to take the first that matches in the same order as the code is looping on items... so "decreasing priority". WDYT?

Reply:
You're the boss!

Reply:
<img src="https://media.giphy.com/media/mLpLzy2XWIN3y/giphy.gif">